### PR TITLE
[kafkareceiver] Add Azure Resource Log Support

### PR DESCRIPTION
**Pull request by cparkins**
**Description:** <Describe what has changed.>

Add support for the Azure Resource Log format so that Diagnostic logs from Azure can be consumed by a Kafka Receiver.

**Link to tracking Issue:** <Issue number if applicable>
#18210 

**Testing:** <Describe what testing was performed and which tests were added.>
Local testing to ensure that data is being converted correctly when Streaming from Azure Diagnostic Settings to Event Hubs and using the Kafka Receiver.

**Documentation:** <Describe the documentation added.>
Added a bullet for the added azure format.

### CODE REVIEW

**Review by atoulme — APPROVED** [review]
LGTM.

**Comment by mx-psi** [comment]
@cparkins can you update to Go 1.21 and run `make gotidy` again? There are some differences in `go mod tidy` between Go 1.20 and Go 1.21 related to `cloud.google.com/metadata`.

**Comment by cparkins** [comment]
@mx-psi I've rebased to remove the modifications made by running make gotidy.

File: receiver/kafkareceiver/azureresourcelogs_unmarshaler.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+package kafkareceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/kafkareceiver"
+
+import (
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/azure"
+)
+
+type AzureResourceLogsUnmarshaler struct {
```
Comment (mx-psi) [inline receiver/kafkareceiver/azureresourcelogs_unmarshaler.go:13]: This should not be exported 
```suggestion
type azureResourceLogsUnmarshaler struct {
```

File: receiver/kafkareceiver/README.md [hunk 2]
```diff
@@ -42,6 +42,7 @@ The following settings can be optionally configured:
   - `raw`: (logs only) the payload's bytes are inserted as the body of a log record.
   - `text`: (logs only) the payload are decoded as text and inserted as the body of a log record. By default, it uses UTF-8 to decode. You can use `text_<ENCODING>`, like `text_utf-8`, `text_shift_jis`, etc., to customize this behavior.
   - `json`: (logs only) the payload is decoded as JSON and inserted as the body of a log record.
+  - `azureresourcelogs`: (logs only) the payload is converted from Azure Resource Logs format to OTel format.
```
Comment (mx-psi) [inline receiver/kafkareceiver/README.md:45]: Should this be `azure_resource_logs` instead for consistency in the casing style?

**Review by mx-psi — APPROVED** [review]

**Pull request merged**